realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -327,6 +327,10 @@ public final void removeChangeListeners() {
      * }
      * </pre>
      *
+     * <p>Note that when the {@link Realm} is accessed from threads other than where it was created,
+     * {@link IllegalStateException} will be thrown. Please avoid using {@code subscribeOn()}, and use
```

Comment:
Imo this still doesn't capture it, pr. my previous review. It is the combination of subscribeOn/observeOn that is problematic.

Also, DynamicRealmObject also have a asObservable method :)